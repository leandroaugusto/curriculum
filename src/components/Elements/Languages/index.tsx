"use client";

import Link from "next/link";
import Image from "next/image";
import { usePathname } from "next/navigation";

import EnFlag from "@/assets/icons/gb-eng.svg";
import BrFlag from "@/assets/icons/br.svg";

export default function Languages(): React.ReactElement<HTMLDivElement> {
  const pathname = usePathname();

  return (
    <nav className="absolute top-4 right-4 md:right-0 print:hidden">
      <ul className="grid gap-3 grid-flow-col text-gray-800">
        <li
          className={`rounded-full overflow-hidden border-2 border-solid ${
            pathname === "/en" ? "border-red-700" : "border-gray-300 scale-90"
          }`}
        >
          <Link href={"/en"}>
            <Image
              alt="English"
              src={EnFlag}
              className="object-cover object-center h-6 w-8 rounded"
            />
          </Link>
        </li>
        <li
          className={`rounded-full overflow-hidden border-2 border-solid ${
            pathname === "/pt" ? "border-blue-800" : "border-gray-300 scale-90"
          }`}
        >
          <Link href={"/pt"}>
            <Image
              src={BrFlag}
              alt="PortuguÃªs"
              className="object-cover object-center h-6 w-8 rounded"
            />
          </Link>
        </li>
      </ul>
    </nav>
  );
}
