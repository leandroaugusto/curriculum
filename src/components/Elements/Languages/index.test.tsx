import { render } from "@testing-library/react";

import Languages from ".";

describe("Languages", () => {
  it("renders the Languages element", () => {
    const { container } = render(<Languages />);
    expect(container).toMatchSnapshot();
  });
});
