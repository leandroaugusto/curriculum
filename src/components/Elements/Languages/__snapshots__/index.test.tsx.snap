// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Languages renders the Languages element 1`] = `
<div>
  <nav
    class="absolute top-4 right-4 md:right-0 print:hidden"
  >
    <ul
      class="flex text-gray-800"
    >
      <li
        class="flex items-center justify-center"
      >
        <a
          href="/en"
        >
          <img
            alt="English"
            class="object-cover object-center h-6 w-8 rounded"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
      </li>
      <li
        class="flex items-center justify-center ml-4"
      >
        <a
          href="/pt"
        >
          <img
            alt="PortuguÃªs"
            class="object-cover object-center h-6 w-8 rounded"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
